client/components/pages/EditDocumentPage.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { Link } from 'react-router';
+import toastr from 'toastr';
+import TinyMCE from 'react-tinymce';
+import FlatButton from 'material-ui/FlatButton';
+import SelectInput from '../forms/SelectInput.jsx';
+import TextInput from '../forms/TextInput.jsx';
+import Sidebar from '../layouts/Sidebar.jsx';
+import * as documentActions from '../../actions/documentActions';
+import handleError from '../../utils/errorHandler';
+
+class EditDocumentPage extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      document: Object.assign({}, props.document),
+      error: {},
+      saving: false
+    };
+    this.onSubmit = this.onSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleEditorChange = this.handleEditorChange.bind(this);
+  }
+
+  componentDidMount() {
+    $('select').material_select();
+    $('#select-box').on('change', this.handleChange);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // debugger;
+    if (this.state.document.id !== nextProps.document.id) {
+      this.setState({ document: Object.assign({}, nextProps.document) });
+    }
+  }
+
+  handleChange(event) {
+    const document = this.state.document;
+    document[event.target.name] = event.target.value.substr(0, 60);
+    // debugger;
+    this.setState({ document });
+  }
+
+  handleEditorChange(event) {
+    const document = this.state.document;
+    document.content = event.target.getContent();
+    // debugger;
+    this.setState({ document });
+  }
+
+  onSubmit(event) {
+    event.preventDefault();
+    this.setState({ saving: true });
+    this.props.actions.updateDocument(this.state.document)
+    .then(() => this.redirect())
+    .catch((error) => {
+      this.setState({ saving: false });
+      handleError(error);
+    });
+  }
+
+  redirect() {
+    this.setState({ saving: false });
+    toastr.success('Document updated successfully');
+    this.context.router.push('/home');
+  }
+
+  render() {
+    return (
+      <div className="new-document-page">
+        <div className="row">
+          <div className="col s12 m4 l3">
+            <Sidebar />
+          </div>
+          <div className="col s12 m8 l9">
+            <div className="create-document container center-align">
+              <h3>Edit Document</h3>
+              <form onSubmit={this.onSubmit}>
+                <div>
+                  <TextInput
+                    name="title"
+                    type="text"
+                    errorText=""
+                    floatText="Title"
+                    handleChange={this.handleChange}
+                    value={this.state.document.title}
```

Comment:
on line 71 you can have:
`const {title, access, content} = this.state.document`